test/action/join.cpp

```diff
@@ -0,0 +1,35 @@
+// Range v3 library
+//
+//  Copyright Eric Niebler 2014
+//
+//  Use, modification and distribution is subject to the
+//  Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at
+//  http://www.boost.org/LICENSE_1_0.txt)
+
+#include <vector>
+#include <range/v3/core.hpp>
+#include <range/v3/action/join.hpp>
+#include <range/v3/algorithm/move.hpp>
+#include <range/v3/algorithm/equal.hpp>
+#include <range/v3/view/transform.hpp>
+#include "../simple_test.hpp"
+#include "../test_utils.hpp"
+
+template<class> struct undef;
```

Comment:
With a more descriptive name - I always call it `show_type` and make it variadic - it would be a good idea to put an intentionally undefined template in e.g. `test_utils`. It would not suck to never have to type this line of code again.